app/src/main/java/org/simple/clinic/questionnaireResponse/QuestionnaireResponse.kt

```diff
@@ -30,4 +36,61 @@ data class QuestionnaireResponse(
     val timestamps: Timestamps,
 
     val syncStatus: SyncStatus
-) : Parcelable
+) : Parcelable {
+  @Dao
+  interface RoomDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun save(questionnaireResponses: List<QuestionnaireResponse>)
+
+    @Query("SELECT * FROM QuestionnaireResponse WHERE uuid = :uuid LIMIT 1")
+    fun getOne(uuid: UUID): QuestionnaireResponse?
+
+    @Query("SELECT * FROM QuestionnaireResponse WHERE questionnaireType = :type")
+    fun getByQuestionnaireType(type: QuestionnaireType): List<QuestionnaireResponse>
+
+    @Update
+    fun updateQuestionnaireResponse(questionnaireResponse: QuestionnaireResponse)
+
+    @Query("SELECT * FROM QuestionnaireResponse WHERE syncStatus = :status")
+    fun withSyncStatus(status: SyncStatus): List<QuestionnaireResponse>
+
+    @Query("""
+      SELECT * FROM QuestionnaireResponse
+      WHERE syncStatus = :syncStatus
+      LIMIT :limit OFFSET :offset
+    """)
+    fun recordsWithSyncStatusBatched(
+        syncStatus: SyncStatus,
+        limit: Int,
+        offset: Int
+    ): List<QuestionnaireResponse>
+
+    @Query("UPDATE QuestionnaireResponse SET syncStatus = :newStatus WHERE syncStatus = :oldStatus")
```

Comment:
I don't fully understand use-case for all the SQL queries here, like this one. But assuming this is boilerplate, we can keep it

Reply:
It's a boilerplate code. Even i don't understand this fully atm.